tools/offline_log_viewer/kafka.py

```diff
@@ -82,7 +90,7 @@ def decode_record(batch, header, record):
     is_ctrl = attrs["control_batch"]
     is_tx_ctrl = is_txn and is_ctrl
     if is_tx_ctrl:
-        record_dict["type"] = self.get_control_record_type(record.key)
+        record_dict["type"] = get_control_record_type(record.key)
```

Comment:
did you also want to remove the duplicate function `get_control_record_type` in `KafkaLog` below?

Reply:
yikes.. I swear I removed it, maybe forgot to commit those changes.. anyway fixed it..

Also added a little regression test for future, had to adjust printing logic to print a wellformed json to stdout.